pkg/controller/daemon/daemon_controller_test.go

```diff
@@ -455,20 +458,15 @@ func TestSimpleDaemonSetLaunchesPods(t *testing.T) {
 // When ScheduleDaemonSetPods is enabled, DaemonSets without node selectors should
 // launch pods on every node by NodeAffinity.
 func TestSimpleDaemonSetScheduleDaemonSetPodsLaunchesPods(t *testing.T) {
```

Comment:
Please use something like this to make sure existing tests still pass when the `ScheduleDaemonSetPods` feature is enabled and disabled:
https://github.com/kubernetes/kubernetes/blob/be11540775e36b1f1acdd6cb43500c1717af0ad4/test/integration/daemonset/daemonset_test.go#L482-L496

Reply:
So, the tests would have failed similar to the ones that failed here https://github.com/kubernetes/kubernetes/pull/67790 , if ScheduleDSPods was enabled(I mean they are running even now, after we enable the feature). But I understand your concern, I will add something similar to what we have in integration tests.